### PR TITLE
fix: radial gradient export alignment and positioning

### PR DESCRIPTION
**Pull request by akshay-gupta7**
### Why does this PR exist?

This PR addresses inconsistencies and missing features in radial gradient handling between Tokens Studio and Figma. Previously, radial gradients were not accurately positioned or aligned during export, and the plugin lacked the ability to match radial gradient tokens against existing Figma paint styles.

### What does this pull request do?

- **Radial Gradient Transformation Engine**:
    - Implemented robust parsing for `radial-gradient` syntax, including support for the `at <position>` keyword and shape/size declarations (`circle`, `ellipse`).
    - Added matrix math to correctly transform CSS coordinates and centers into Figma's relative transformation space, ensuring pixel-perfect alignment.
- **Paint Style Matcher Support**:
    - Added support for `GRADIENT_RADIAL` in `paintStyleMatchesColorToken.ts`. This allows the plugin to correctly identify when a Figma style already matches a radial gradient token, preventing unnecessary style overrides.
- **Hardened Parsing & Modern Color Support**:
    - Refactored gradient part detection to use keyword-based identification (`at`, `circle`, etc.). This fixes a previous bug where modern color formats like `oklch()` or named colors were sometimes misclassified as positioning metadata.
    - Added non-gradient guards to prevent the parser from attempting to process solid colors during style matching.
- **Types & Cleanup**:
    - Removed redundant type fallbacks and non-null assertions across `gradients.ts` and `setColorValuesOnTarget.ts`.
    - Fixed documentation typos in transformation comments.
- **Comprehensive Testing**:
    - Added new test suites covering radial and conic gradients, including edge cases with non-standard rotations and centers.

### Testing this change

- **Automated Tests**:
    - Run `yarn workspace @tokens-studio/figma-plugin test src/plugin/figmaUtils/styleMatchers/paintStyleMatchesColorToken.test.ts` to verify radial style matching.
    - Run `yarn workspace @tokens-studio/figma-plugin test src/plugin/figmaTransforms/gradients.test.ts` to verify the transformation engine.
- **Manual Verification**:
    - Apply a radial gradient token with a custom position (e.g., `radial-gradient(at top left, #fff, #000)`) to a layer and verify it matches the expected visual output in Figma.

### Additional Notes (if any)

This PR also includes updates to the `PaintStyle` test mock to align with the latest `@figma/plugin-typings`.

### CODE REVIEW

File: packages/tokens-studio-for-figma/src/plugin/figmaUtils/styleMatchers/paintStyleMatchesColorToken.ts [hunk 1]
```diff
@@ -2,14 +2,19 @@ import { isPaintEqual } from '@/utils/isPaintEqual';
 import { convertStringToFigmaGradient } from '../../figmaTransforms/gradients';
 import { convertToFigmaColor } from '../../figmaTransforms/colors';
 
+// Helper function to check if a value is any type of gradient
+const isGradient = (value: string): boolean => value?.startsWith?.('linear-gradient')
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/plugin/figmaUtils/styleMatchers/paintStyleMatchesColorToken.ts:6]: do we support repeating-linear-gradient etc? you added that further above, but its missing from here

Comment (akshay-gupta7) [reply packages/tokens-studio-for-figma/src/plugin/figmaUtils/styleMatchers/paintStyleMatchesColorToken.ts:6]: supported in the token, not in the export yet, removed it in [remove repeating gradient check](https://github.com/tokens-studio/figma-plugin/pull/3837/commits/a3d850658403ab2d9008b0bb6afaacdfb863bebe)

Comment (akshay-gupta7) [reply packages/tokens-studio-for-figma/src/plugin/figmaUtils/styleMatchers/paintStyleMatchesColorToken.ts:6]: will raise a separate PR to include support for exporting repeating gradients

**Review by six7 — APPROVED** [review]
failing CI but approved otherwise
